components/StreamField/StreamField.tsx

```diff
@@ -0,0 +1,39 @@
+import { SectionBlock, IssuesBlock, NewsBlock, TopicsBlock } from './blocks';
+import type { StreamFieldProps } from './types';
+import styles from './StreamField.module.scss';
+
+const BLOCKS = {
+    section: SectionBlock,
+    issues: IssuesBlock,
+    news: NewsBlock,
+} as const;
+type BlockType = keyof typeof BLOCKS;
+
+const StreamField = ({ body, topics }: StreamFieldProps) => {
+    if (!body) {
+        return null;
+    }
+
+    return (
+        <div className={styles.container}>
+            {body.map((block, i) => {
+                const blockType = block.type;
+                if (blockType === 'topics') {
+                    return (
+                        <div key={i} className={styles.section}>
+                            <TopicsBlock {...block.value} topics={topics} />
+                        </div>
+                    );
+                }
+                const Block = BLOCKS[blockType as BlockType];
```

Comment:
This was a bit confusing to be honest. Maybe it can be `const BlockComponent = ...` instead?